integration-tests/src/test_loop/tests/epoch_sync.rs

```diff
@@ -0,0 +1,188 @@
+        .genesis(genesis.clone())
+        .clients(clients)
+        .stores_override(stores)
+        .test_loop_data_dir(tempdir)
+        .config_modifier(|config, _| {
+            // Enable epoch sync, and make the horizon small enough to trigger it.
+            config.epoch_sync.enabled = true;
+            config.epoch_sync.epoch_sync_horizon = 30;
+            config.epoch_sync.epoch_sync_accept_proof_max_horizon = 20;
+            // Make header sync horizon small enough to trigger it.
+            config.block_header_fetch_horizon = 8;
+            // Make block sync horizon small enough to trigger it.
+            config.block_fetch_horizon = 3;
+        })
+        .skip_warmup()
+        .build();
+
+    // Note: TestLoopEnv does not currently propagate the network info to other peers. This is because
+    // the networking layer is completely mocked out. So in order to allow the new node to sync, we
+    // need to manually propagate the network info to the new node. In this case we'll tell the new
+    // node that node 0 is available to sync from.
+    let chain0 = &test_loop.data.get(&node_datas[0].client_sender.actor_handle()).client.chain;
+    let peer_info = HighestHeightPeerInfo {
+        archival: false,
+        genesis_id: GenesisId { chain_id: genesis.config.chain_id, hash: *chain0.genesis().hash() },
+        highest_block_hash: chain0.head().unwrap().last_block_hash,
+        highest_block_height: chain0.head().unwrap().height,
+        tracked_shards: vec![],
+        peer_info: PeerInfo {
+            account_id: Some(accounts[0].clone()),
+            addr: None,
+            id: node_datas[0].peer_id.clone(),
+        },
+    };
+    node_datas[NUM_CLIENTS].client_sender.send(SetNetworkInfo(NetworkInfo {
+        connected_peers: Vec::new(),
+        highest_height_peers: vec![peer_info], // only this field matters.
+        known_producers: vec![],
+        num_connected_peers: 0,
+        peer_max_count: 0,
+        received_bytes_per_sec: 0,
+        sent_bytes_per_sec: 0,
+        tier1_accounts_data: Vec::new(),
+        tier1_accounts_keys: Vec::new(),
+        tier1_connections: Vec::new(),
+    }));
+
+    // Check that the new node will reach a high height as well.
+    let new_node = node_datas.last().unwrap().client_sender.actor_handle();
+    test_loop.set_every_event_callback(move |test_loop_data| {
+        let client = &test_loop_data.get(&new_node).client;
+        let header_head_height = client.chain.header_head().unwrap().height;
+        let head_height = client.chain.head().unwrap().height;
+        tracing::info!(
+            "New node sync status: {:?}, header head height: {:?}, head height: {:?}",
+            client.sync_status,
+            header_head_height,
+            head_height
+        );
+    });
+    let new_node = node_datas.last().unwrap().client_sender.actor_handle();
+    let node0 = node_datas[0].client_sender.actor_handle();
+    test_loop.run_until(
```

Comment:
Are we checking that the sync happened through epoch_sync and not something like block sync? Should we be monitoring the sync status somewhere?

Reply:
Ah yeah... we should do that somehow... How would that work hmm...